cmd/root.go

```diff
@@ -73,60 +73,142 @@ var (
-			// below, we're stripping any colors from the output after we've
-			// added them. The problem is that, besides being very inefficient,
-			// this actually also strips other special characters from the
-			// intended output, like the progressbar formatting ones, which
-			// would otherwise be fine (in a TTY).
-			//
-			// It would be much better if we avoid messing with the output and
-			// instead have a parametrized instance of the color library. It
-			// will return colored output if colors are enabled and simply
-			// return the passed input as-is (i.e. be a noop) if colors are
-			// disabled...
-			stdout.Writer = colorable.NewNonColorable(os.Stdout)
-			stderr.Writer = colorable.NewNonColorable(os.Stderr)
-		}
-		stdlog.SetOutput(logger.Writer())
-		logger.Debugf("k6 version: v%s", consts.FullVersion())
-		return nil
-	},
+			if noColor {
+				// TODO: figure out something else... currently, with the wrappers
+				// below, we're stripping any colors from the output after we've
+				// added them. The problem is that, besides being very inefficient,
+				// this actually also strips other special characters from the
+				// intended output, like the progressbar formatting ones, which
+				// would otherwise be fine (in a TTY).
+				//
+				// It would be much better if we avoid messing with the output and
+				// instead have a parametrized instance of the color library. It
+				// will return colored output if colors are enabled and simply
+				// return the passed input as-is (i.e. be a noop) if colors are
+				// disabled...
+				stdout.Writer = colorable.NewNonColorable(os.Stdout)
+				stderr.Writer = colorable.NewNonColorable(os.Stderr)
+			}
+			stdlog.SetOutput(logger.Writer())
+			logger.Debugf("k6 version: v%s", consts.FullVersion())
+			return nil
+		},
+	}
+	return RootCmd
 }
 
 // Execute adds all child commands to the root command sets flags appropriately.
 // This is called by main.main(). It only needs to happen once to the rootCmd.
-func Execute() {
+func Execute() { //nolint:funlen
+	ctx := context.Background()
+	logger := &logrus.Logger{
+		Out:       os.Stderr,
+		Formatter: new(logrus.TextFormatter),
+		Hooks:     make(logrus.LevelHooks),
+		Level:     logrus.InfoLevel,
+	}
+
+	var fallbackLogger logrus.FieldLogger = &logrus.Logger{
+		Out:       os.Stderr,
+		Formatter: new(logrus.TextFormatter),
+		Hooks:     make(logrus.LevelHooks),
+		Level:     logrus.InfoLevel,
+	}
+	RootCmd := getRootCmd(ctx, logger, fallbackLogger)
+	confDir, err := os.UserConfigDir()
+	if err != nil {
+		logrus.WithError(err).Warn("could not get config directory")
+		confDir = ".config"
+	}
+	defaultConfigFilePath = filepath.Join(
+		confDir,
+		"loadimpact",
+		"k6",
+		defaultConfigFileName,
+	)
+
+	RootCmd.PersistentFlags().AddFlagSet(rootCmdPersistentFlagSet())
+
+	archiveCmd := getArchiveCmd(logger)
+	RootCmd.AddCommand(archiveCmd)
+	archiveCmd.Flags().SortFlags = false
+	archiveCmd.Flags().AddFlagSet(archiveCmdFlagSet())
```

Comment:
hmm instead of building all of these flags here, is there a reason you can't add the flag sets in the respective commands? For example, 
```
	archiveCmd.Flags().SortFlags = false
	archiveCmd.Flags().AddFlagSet(archiveCmdFlagSet())
```
can occur here, right? https://github.com/loadimpact/k6/blob/6475b7aad50b49bcb4b4d9bcc589c3624b4458a2/cmd/archive.go#L100-L101

That way the diff will be even simpler - just moving these commands out of the magical `init()` to the respective `getFooCommand()`

Reply:
I needed to that for the `convert` one ... but didn't do it for all the others just so that there were less changes as now and before (except for `convert`) the Command is first added and than it's flags are being added. 

AFAIK there is no difference and the `convert` tests pass so :shrug: 

Arguably this needs to happen IMO so I am fine with doing it